lib/std/os.zig

```diff
@@ -42,17 +42,15 @@ test {
 }
 
 /// See also `getenv`. Populated by startup code before main().
-/// TODO this is a footgun because the value will be undefined when using `zig build-lib`.
-/// https://github.com/ziglang/zig/issues/4524
-pub var environ: [][*:0]u8 = undefined;
+pub var environ: ?[][*:0]u8 = null;
```

Comment:
As noted [here](https://github.com/ziglang/zig/issues/23372#issuecomment-2755734684) this breaks ~800 easily observable usages.